src/main/java/org/jitsi/srtp/crypto/Aes.java

```diff
@@ -335,6 +348,19 @@ private static CipherFactory benchmark(
                 else if (t instanceof ThreadDeath)
                     throw (ThreadDeath) t;
             }
+
+            if (chosenFactoryClass)
+            {
+                if (minFactory != null)
```

Comment:
I'm not sure I understand this well. If this loop iteration is for the user-configured factory class, return the fastest factory. If a previous non-user-configured factory was faster, we short-circuit and return it (the faster one)?

Should that return `factory` instead of `minFactory`?

Reply:
Well, elsewhere in the code I reorganize the list so the user-configured factory class is always first, so we avoid profiling factories we don't need to.  But I agree that's fragile and confusing.

I've reorganized it now, is it clearer?